### PR TITLE
Add CreateClaimsPrincipalCrossVersionTest test case

### PR DESCRIPTION
**Pull request by ChenYan98**
#668

### CODE REVIEW

File: test/CrossVersionTokenValidation.Tests/CompareContext.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.CrossVersionTokenValidation.Tests
```
Comment (brentschmaltz) [inline test/CrossVersionTokenValidation.Tests/CompareContext.cs]: We can link to this file and #ifdef out any code that we can't use.

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSerializer.cs [hunk 2]
```diff
@@ -318,20 +318,24 @@ public virtual SamlAttribute ReadAttribute(XmlDictionaryReader reader)
...
-
-                var nameSpace = reader.GetAttribute(SamlConstants.Attributes.AttributeNamespace, null);
-                if (string.IsNullOrEmpty(nameSpace))
+                attribute.Namespace = reader.GetAttribute(SamlConstants.Attributes.AttributeNamespace, null);
+                if (string.IsNullOrEmpty(attribute.Namespace))
                     throw LogReadException(LogMessages.IDX11115, SamlConstants.Elements.Attribute, SamlConstants.Attributes.AttributeNamespace);
 
-                attribute.Namespace = nameSpace;
-
                 // TODO is this the right thing?
```
[the remark this answers is not in the document]
Comment (ChenYan98) [reply src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSerializer.cs]: Fixed.

File: src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlConstants.cs [hunk 3]
```diff
@@ -48,6 +48,8 @@ public static class SamlConstants
         public const string UserName = "UserName";
         public const string UserNameNamespace = "urn:oasis:names:tc:SAML:1.1:nameid-format:WindowsDomainQualifiedName";
         public const string UnspecifiedAuthenticationMethod = "urn:oasis:names:tc:SAML:1.0:am:unspecified";
+        public const string ClaimType2009Namespace = "http://schemas.xmlsoap.org/ws/2009/09/identity/claims";
```
[the remark this answers is not in the document]
Comment (ChenYan98) [reply src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlConstants.cs]: Switch to internal.

File: src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs [hunk 4]
```diff
@@ -118,7 +118,7 @@ public class TokenValidationParameters
         /// <summary>
         /// This is the fallback authenticationtype that a <see cref="ISecurityTokenValidator"/> will use if nothing is set.
         /// </summary>
-        public static readonly string DefaultAuthenticationType = "AuthenticationTypes.Federation";
+        public static readonly string DefaultAuthenticationType = "Federation";
```
[the remark this answers is not in the document]
Comment (ChenYan98) [reply src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs]: Add note.

File: test/CrossVersionTokenValidation.Tests/CompareContext.cs (hunk 1 above)
Comment (ChenYan98) [reply test/CrossVersionTokenValidation.Tests/CompareContext.cs]: Fixed

**Comment by brentschmaltz** [comment]
:shipit:

**Pull request merged**